### PR TITLE
[FloatingActionButtonLocation] Add diagrams to documentation of FloatingActionButtonLocation

### LINKED ISSUE #24732: FloatingActionButtonLocation needs diagrams
Especially showing all the various predefined values

### PR DESCRIPTION
**Pull request by royarg02**
## Description
This PR adds diagrams showing the usage of **FloatingActionButtonLocation**.

The linked diagrams in the docs will be available once https://github.com/flutter/assets-for-api-docs/pull/123 is merged.

## Related Issues
Fixes #24732.

## Tests
No tests, since it is just some additions to the documentation.

## Checklist
Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change
Did any tests fail when you ran them? Please read [Handling breaking changes].
- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/material/floating_action_button_location.dart [hunk 1]
```diff
@@ -58,6 +58,64 @@ const double kMiniButtonOffsetAdjustment = 4.0;
 /// [FloatingActionButton] placements in Material Design applications. These
 /// locations are available as static members of this class.
 ///
+/// ## Floating Action Button placements
+///
+/// The following diagrams show the available common placement locations for the FloatingActionButton.
```
Comment (Piinks) [inline packages/flutter/lib/src/material/floating_action_button_location.dart:63]:
```suggestion
/// The following diagrams show the available placement locations for the FloatingActionButton.
```

**Review by Piinks — COMMENTED** [review]
Hi @RoyARG02 thanks for the contribution! Just a small nit below.

File: packages/flutter/lib/src/material/floating_action_button_location.dart (hunk 1 above)
Comment (RoyARG02) [reply packages/flutter/lib/src/material/floating_action_button_location.dart]: I fixed it.

**Review by Piinks — APPROVED** [review]
LGTM! Thank you!

**Comment by Piinks** [comment]
@RoyARG02 can you update your branch with the latest from master? The failing checks here are not caused by your PR.

**Comment by RoyARG02** [comment]
@Piinks I've updated the branch.

**Comment by RoyARG02** [comment]
Hey @Piinks sorry to bug you, but the docs aren't showing as expected in the website:

fab_location_doc

**Pull request merged**